cmd/deploy_test.go

```diff
@@ -0,0 +1,209 @@
+			err2 := testClient.CreateOrUpdateNamespace("foo", map[string]string{}, map[string]string{})
+			if err2 != nil {
+				fmt.Print("ERROR!", err2)
+				return
+			}
+
+			err := testClient.Apply("foo", &v1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{Kind: "ConfigMap", APIVersion: "v1"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "bar",
+					Namespace: "foo",
+				},
+				Data: map[string]string{"foo": "lala"},
+			})
+			if err != nil {
+				fmt.Print("ERROR!", err)
+				return
+			}
+
+			_, value, _ := testClient.GetEnvValue(config.Hooks["elastic"].Before, "foo")
+			err3 := testClient.ApplyText("foo", value)
+			if err3 != nil {
+				fmt.Print("ERROR!", err3)
+				return
+			}
+			assert.Equal(t, value, 1, "Output")
+			_ = testEnv.Stop()
+		})
+	}
+}
+
+func Test_deployPhase(t *testing.T) {
+	type args struct {
+		plaformConfig types.PlatformConfig
+		phase         string
+		fn            order.DeployFn
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "basic",
+			args: args{
+				plaformConfig: types.PlatformConfig{},
+				phase:         "crds",
+				fn:            order.PhasesExtra[order.Crds].Fn,
+			},
+			want: true,
+		},
+		{
+			name: "pre",
+			args: args{
+				plaformConfig: types.PlatformConfig{
+					Hooks: map[string]types.Hook{
+						"crds": types.Hook{
+							Before: kommons.EnvVar{
+								Value: string([]byte(`{
+								    "kind": "Namespace",
+								    "apiVersion": "v1",
+								    "metadata": {
+								        "name": "test"
+								    }
+								}`)),
+							},
+						},
+					},
+				},
+				phase: "crds",
+				fn:    order.PhasesExtra[order.Crds].Fn,
+			},
+			want: true,
+		},
+		{
+			name: "post",
+			args: args{
+				plaformConfig: types.PlatformConfig{
+					Hooks: map[string]types.Hook{
+						"crds": types.Hook{
+							After: kommons.EnvVar{
+								Value: string([]byte(`{
+								    "kind": "Namespace",
+								    "apiVersion": "v1",
+								    "metadata": {
+								        "name": "test"
+								    }
+								}`)),
+							},
+						},
+					},
+				},
+				phase: "crds",
+				fn:    order.PhasesExtra[order.Crds].Fn,
+			},
+			want: true,
+		},
+		{
+			name: "both",
+			args: args{
+				plaformConfig: types.PlatformConfig{
+					Hooks: map[string]types.Hook{
+						"crds": types.Hook{
+							Before: kommons.EnvVar{
+								Value: string([]byte(`{
+								    "kind": "Namespace",
+								    "apiVersion": "v1",
+								    "metadata": {
+								        "name": "testbefore"
+								    }
+								}`)),
+							},
+							After: kommons.EnvVar{
+								Value: string([]byte(`{
+								    "kind": "Namespace",
+								    "apiVersion": "v1",
+								    "metadata": {
+								        "name": "testafter"
+								    }
+								}`)),
+							},
+						},
+					},
+				},
+				phase: "crds",
+				fn:    order.PhasesExtra[order.Crds].Fn,
+			},
+			want: true,
+		},
+	}
+	config, bin, err := testenv.StartTestEnv("1.19.2")
```

Comment:
In the current set up, the tests are stateful. We need a way of doing correct setup and teardown per file, if not per loop.